File: December2021/Smallest Integer Divisible by K.cpp
class Solution {
public: 
    int smallestRepunitDivByK(int K) {
        for (int r = 0, N = 1; N <= K; ++N)
            if ((r = (r * 10 + 1) % K) == 0)
                return N;
        return -1;
    }
};

File: December2021/SmallestIntegerDivisiblebyK.java
class Solution {
	// TC : O(K)
	// SC : O(1)
	public int smallestRepunitDivByK(int k) {
		if (k % 2 == 0 || k % 5 == 0) return -1;
		int remainder = 0;
		for (int counter = 1; counter <= k; counter++) {
			remainder = (remainder * 10 + 1) % k;
			if (remainder == 0) return counter;
		}
		return -1;
	}
}